docs/guides/readme.md

```diff
@@ -6,6 +6,7 @@ This section contains applications, and scenario specific guides for Pomerium.
 - The [argo](./argo.md) guide demonstrates how Pomerium can be used to add access control to [Argo](https://argoproj.github.io/projects/argo).
 - The [Cloud Run](./cloud-run.md) recipe demonstrates deploying Pomerium to Google Cloud Run as well as using it to Authorize users to protected Cloud Run endpoints.
 - The [code-server](./code-server.md) guide demonstrates how Pomerium can be used to add access control to third-party applications that don't ship with [fine-grained access control](https://github.com/cdr/code-server/issues/905). code-server is a tool to run Visual Studio code as a web application.
+- See [Enroll a Device](./enroll-device.md) to learn how to register a secure enclave (TPM, Yubikey, etc) to access routes requiring a device ID.
```

Comment:
I don't think a Yubikey is a secure enclave, though I'm not sure what the term specifically refers to. Google turns up Apple-related links, so I guess its a term they use?

In WebAuthn terminology there are two "Authenticator Attachment Modalities": "Platform" authenticators (or Internal authenticators), which are physically bound to a specific device and "Cross-Platform" authenticators (or Roaming authenticators) which are removable from a specific device and can "roam" between devices.

If "Secure Enclave" `==` "Platform", then it shouldn't include Yubikey.

Reply:
CC @desimone to weigh in on terminology to use here. This relates to a slack conversation we had on terminology. I'll link to this in the thread.

Reply:
The current state of hardware backed attestation devices is confusing, especially in the context of the WebAuthN. Every hardware provider brands their approach which certainly doesn't help. Apple calls their Secure Enclave. Google brands anything remotely TPM related Titan. Intel calls theirs SGX and so on. And the line between each technology type is very gray. The spec's terminology is most precise, but is not well known or used outside of webauthn so I've been searching for reasonable terminology that: is understandable to a more general audience, technically correct to the spec. 

> In WebAuthn terminology there are two "Authenticator Attachment Modalities": "Platform" authenticators (or Internal authenticators), which are physically bound to a specific device and "Cross-Platform" authenticators (or Roaming authenticators) which are removable from a specific device and can "roam" between devices.

As @calebdoxsey mentioned, in webauthn-land we basically have two big category of hardware based devices: **platform**, and **cross-platform** where the big differentiator from the is whether or not the hardware can "roam."

I suggest.

**Trusted Execution Environment** (TEE) or to refer to all hardware backed methods including secure enclaves, and hardware security keys. 

**Secure Enclave** to refer to all **platform** based methods. Admittedly, I don't love calling it that because as far as I can tell this is mostly the apple specific implementation but in common parlance it's what I hear used most often when people talk about the general category.

**Hardware security key** to refer to **cross-platform** / **roaming** capable hardware. I don't love this either, but it does imply that the device can be used, and moved like a physical key.

Very open to ideas though.

- https://cryptologie.net/article/501/hardware-solutions-to-highly-adversarial-environments-part-3-trusted-execution-environment-tee-sgx-trustzone-and-hardware-security-tokens/
- https://developers.yubico.com/WebAuthn/WebAuthn_Developer_Guide/Platform_vs_Cross-Platform.html

<img width="788" alt="Screen Shot 2021-12-20 at 3 20 41 PM" src="https://user-images.githubusercontent.com/1544881/146845497-e4f73a56-f200-4965-b501-9575465feeb3.png">

Reply:
Thanks for the discussion, y'all. Adjusted in https://github.com/pomerium/pomerium/pull/2830/commits/88b9331976c1533d24b4605e99040a6d4727e020